backend/sender.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pickle as pkl
+import signal
+import os
+
+from py_utils.wait.BooleanWaitHandler import BooleanWaitHandler
+
+from amlip_py.node.ModelManagerSenderNode import ModelManagerSenderNode, ModelReplier
+from amlip_py.types.AmlipIdDataType import AmlipIdDataType
+from amlip_py.types.ModelReplyDataType import ModelReplyDataType
+from amlip_py.types.ModelRequestDataType import ModelRequestDataType
+
+# Domain ID
+DOMAIN_ID = 166
+
+# Variable to wait to the model request
+waiter = BooleanWaitHandler(True, False)
+
+
+# Custom model replier
+class CustomModelReplier(ModelReplier):
+
+    def fetch_model(
+            self,
+            request: ModelRequestDataType) -> ModelReplyDataType:
+
+        print('Request received:\n')
+        print(request.to_string())
+        print('\n')
+
+        # Get the path to the Downloads directory
+        downloads_path = os.path.join(os.path.expanduser("~"), "Downloads")
+
+        # Create the full path to the file
+        model_path = os.path.join(downloads_path, "model_.json")
+
+        try:
+            with open(model_path, 'r') as file:
+                file_data = file.read()
+            reply = ModelReplyDataType(file_data)
+            print('Publish reply\n')
+            return reply
+        except FileNotFoundError:
+            print("File not found.")
+            return None
+
+
+def main():
+    """Execute main routine."""
+
+    # Create ID
+    id_sender = AmlipIdDataType('ModelManagerSender')
+
+    # Create node
+    model_sender_node = ModelManagerSenderNode(
+        id=id_sender,
+        domain=DOMAIN_ID)
+
+    print(f'Node created: {model_sender_node.get_id()}. '
+          'Already processing models.')
+
+    # e.g. statistics data
+    data = {
+        'name': 'MobileNet V1',
+        'size': 56
+    }
+
+    # Start node
+    model_sender_node.start(
```

Comment:
I don’t think it makes sense to assume that a model sender node could be operating without statistics data, but if statistics are missing, the only issue would be that the node would be operating without any tasks because no one would request the model if it hasn't sent any statistics.